rebasehelper/specfile.py

```diff
@@ -410,14 +395,16 @@ def write_updated_patches(self, patches, disable_inapplicable):
                     patch_inapplicable = [x for x in patches['inapplicable'] if patch_name in x]
                 else:
                     patch_inapplicable = None
-                if check_not_applied or patch_removed:
-                    self.spec_content[index] = '#{0} {1}\n'.format(' '.join(fields[:-1]),
-                                                                   os.path.basename(patch_name))
+                if patch_inapplicable or patch_removed:
+                    if disable_inapplicable:
+                        self.spec_content[index] = '#{0} {1}\n'.format(' '.join(fields[:-1]),
+                                                                       os.path.basename(patch_name))
+                    inapplicable_patches.append(patch_num)
```

Comment:
This doesn't seem right, it will append removed patches to `inapplicable_patches` list.
Also `check_not_applied` is now ignored completely.